identity/src/main/java/com/android/identity/trustmanagement/TrustManager.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.identity.trustmanagement
+
+import com.android.identity.storage.StorageEngine
+import org.bouncycastle.asn1.x500.X500Name
+import java.io.ByteArrayInputStream
+import java.security.cert.CertificateFactory
+import java.security.cert.PKIXCertPathChecker
+import java.security.cert.X509Certificate
+
+/**
+ * This class is used for the verification of a certificate chain
+ */
+class TrustManager(val certificateStorage: StorageEngine) {
```

Comment:
I think perhaps I wasn't clear enough when we discussed this. What I meant with the second bullet in Issue #308 - *"The app will have full control of of which public keys it trusts, initially just hardcode this with the keys we already trust, e.g. keys and certificates from appverifier/src/main/res/raw/"* - was the following: I don't want the library code to be managing its own files, instead I want the application to be feeding the library code with whatever certificates (and VICALs) that it decides to use. That way it's up to the application how this is stored... for example, some certificates might be hardcoded constants, others might be loaded from disk, and some might be retrieved from the network.

So, I would expect `appverifier` to use it like this

```
class VerifierApp : Application() {

  [...]

  private lateinit var userCertStorage: StorageEngine
  private lateinit var trustManager: TrustManager

  [...]

  init {
    trustManager = TrustManager()
    trustManager.addCertificate(getCertificateFromBlob(context, R.raw.apple_iaca));
    // [...] repeat for all the other hard-coded IACA certificates we have in appverifier/src/main/res/raw/

    // Add certificates configured by the user
    userCertStorage = AndroidStorageEngine(context, userCertPath, false)
    for (key in userCertStorage.enumerate()) {
      encodedCert = userCertStorage.get(key)
      trustManager.addCertificate(getCertificateFromBlob(context, encodedCert));
    }
  } 
}
```

This code uses an `AndroidStorageEngine` for the persistence but you could also use something else in the app. Either way, the point is that the library isn't concerned with where the certificates come from.

I think this will also make the tests simpler to read.